File: backend/src/main/java/com/nlu/WebThuongMai/dto/request/orderReq/OrderCreateRequest.java
package com.nlu.WebThuongMai.dto.request.orderReq;

import jakarta.validation.constraints.NotNull;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.experimental.FieldDefaults;

import java.time.LocalDateTime;
import java.util.Set;

@Data
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class OrderCreateRequest {
    @NotNull
    long userId;
    String note;
    LocalDateTime createdDate;
    String address;
    Set<OrderItemRequest> orderItems;
    CouponRequest coupon;
}

File: frontend/src/app/core/models/request/order/OrderCreateRequest.ts
import { CouponRequest } from "../coupon/CouponRequest";
import { OrderItemRequest } from "./OrderItemRequest";

export interface OrderCreateRequest {
    userId: number;
	note?: string;
	address: string;
	createdDate?: string; 
	orderItems: OrderItemRequest[];
	coupon?: CouponRequest;
}
